src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,35 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate startDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate endDate = LocalDate.parse(dateTo, formatter);
+
+        StringBuilder report = new StringBuilder();
+        report.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+
+        for (String name : names) {
+            int salary = 0;
+            for (String line : data) {
+                String[] parts = line.split(" ");
+                if (parts[1].equals(name)) {
+                    LocalDate workDate = LocalDate.parse(parts[0], formatter);
+                    if (workDate.isAfter(startDate.minusDays(1))
```

Comment:
You could use compareTo to compare dates directly. This is more straightforward and reduces the risk of off-by-one errors with minusDays and plusDays. e.g
`if (workDate.compareTo(startDate) >= 0 && workDate.compareTo(endDate) <= 0)`
I would also extract the check to a method e.g. 'isDateWithinRange'

Reply:
Please consider the comment